src/server/scripts/Spells/spell_shaman.cpp

```diff
@@ -213,6 +214,48 @@ class spell_sha_ancestral_guidance_heal : public SpellScript
     }
 };
 
+// 114052 - Ascendance (Restoration)
+class spell_sha_ascendance_restoration : public AuraScript
+{
+    PrepareAuraScript(spell_sha_ascendance_restoration);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_SHAMAN_RESTORATIVE_MISTS });
+    }
+
+    bool CheckProc(ProcEventInfo& procInfo)
+    {
+        if (!procInfo.GetHealInfo() || !procInfo.GetSpellInfo())
+            return false;
+
+        if (procInfo.GetSpellInfo()->Id == SPELL_SHAMAN_RESTORATIVE_MISTS)
+            return false;
+
+        if (!procInfo.GetHealInfo()->GetHeal())
+            return false;
+
+        return true;
+    }
+
+    void OnProc(AuraEffect* aurEff, ProcEventInfo& procInfo)
+    {
+        int32 heal = procInfo.GetHealInfo()->GetHeal();
+        if (heal > 0)
```

Comment:
Ask @Shauren if it's best to use heal > 0 get if GetHeal is EffectiveHeal (not sure if this accounts for anything that might get absorbed, we now handle heal absorbs).

Reply:
GetHeal() is not containing the absorbed heal, e.g. by spell 365153

ref
```cpp
// this is called in void Unit::CalcHealAbsorb
void HealInfo::AbsorbHeal(uint32 amount)
{
    amount = std::min(amount, GetHeal());
    _absorb += amount;
    _heal -= amount;
    amount = std::min(amount, GetEffectiveHeal());
    _effectiveHeal -= amount;
    _hitMask |= PROC_HIT_ABSORB;
}
```

Reply:
Yeah, we need tests and numbers to see.

Reply:
on retail is healing the absorb amount, with last commit works like retail (tested both with same spell)